c10/xpu/XPUFunctions.cpp

```diff
@@ -39,15 +40,20 @@ void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
     auto device_list = platform.get_devices();
     for (const auto& device : device_list) {
       if (device.is_gpu()) {
-        devices.push_back(std::make_unique<sycl::device>(device));
+        gDevicePool.devices.push_back(std::make_unique<sycl::device>(device));
+        if (device.has(sycl::aspect::ext_intel_device_info_uuid))
+          gDevicePool.uuids.push_back(
+              device.get_info<sycl::ext::intel::info::device::uuid>());
+        else
+          gDevicePool.uuids.push_back(std::array<unsigned char, 16>{});
       }
     }
   }
 }
 
 inline void initGlobalDevicePoolState() {
   // Enumerate all GPU devices and record them.
-  enumDevices(gDevicePool.devices);
+  enumDevices();
```

Comment:
why we remove the input parameter.
If not needed, perhaps it's better to inline the whole func logic here?

Reply:
we need to save the uuid into the gDevicePool when enumerating the devices.
we can inline the function here.